### PR TITLE
Add telepresence tasks to Makefile

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

### Description:

I added Makefile tasks to run telepresence.

You can replace deployment/vald-gateway on your k8s clusters just by running:

```sh
make telepresence/swap/gateway SWAP_IMAGE=vdaas/vald-gateway SWAP_TAG=latest
```

statefulset and daemonset will also be swappable after telepresence 0.105 released.
ref.) https://github.com/telepresenceio/telepresence/commit/bb7473fbf19ed4f61796a5e32747e23de6ab03da

```sh
make telepresence/swap/agent-ngt SWAP_DEPLOYMENT_TYPE=statefulset SWAP_IMAGE=vdaas/vald-agent-ngt SWAP_TAG=latest
```

### Related Issue:

Nothing.

### How Has This Been Tested?:

In my local KinD.

### Environment:

- Golang Version: 1.14
- Docker Version: 19.03.5
- Kubernetes Version: 1.16.3
- NGT Version: 1.9.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [X] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensure all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kpango** [comment]
@rinx  LGTM, for more improvement,
in many case local build image is latest or nightly, so please add default variable if  SWAP_IMAGE & SWAP_TAG is not set

**Comment by rinx** [comment]
default variables here
https://github.com/vdaas/vald/pull/285/files#diff-b67911656ef5d18c4ae36cb6741b7965R48-R50
however the default image is just only vald-gateway.

**Comment by kpango** [comment]
I saw that, I mean vald has a lot of component and it will be increase may be no one can remember each components name.
I think it's better to check ENV var is nil or not before teleplesence execution

**Comment by rinx** [comment]
I've just added default values for each components.
@kpango Please review :pray:

File: Makefile.d/functions.mk [hunk 1]
```diff
@@ -79,3 +79,13 @@ define go-lint
...
 	golangci-lint run --enable-all --disable=gochecknoglobals --fix --color always -j 16 --skip-dirs apis/grpc --exclude-use-default=false ./...
 endef
+
+define telepresence
+	[ -z $(SWAP_IMAGE) ] && IMAGE=$2 || IMAGE=$(SWAP_IMAGE) \
+	&& echo "telepresence replaces $(SWAP_DEPLOYMENT_TYPE)/$1 with $${IMAGE}:$(SWAP_TAG)" \
+	&& telepresence \
+	    --swap-deployment $1 \
+	    --docker-run --rm -it $${IMAGE}:$(SWAP_TAG)
+	    ## will be available after telepresence 0.105 released
```
Comment (kpango) [inline Makefile.d/functions.mk]: nice

**Pull request merged**